terraform/slurm_cluster/modules/slurm_controller_instance/main.tf

```diff
@@ -116,6 +116,41 @@ data "local_file" "cgroup_conf_tpl" {
   filename = local.cgroup_conf_tpl
 }
 
+###############################
+# DATA: PROLOG/EPILOG SCRIPTS #
+###############################
+
+data "local_file" "external_epilog" {
+  filename = "${path.module}/files/external_epilog.sh"
+}
+
+data "local_file" "external_prolog" {
+  filename = "${path.module}/files/external_prolog.sh"
+}
+
+data "local_file" "setup_external" {
+  filename = "${path.module}/files/setup_external.sh"
+}
```

Comment:
What is the advantage of using local_file data vs function like file? My assumption is that error/graph are more better.

Reply:
Mostly I am mimicking existing pattern in the code, but yes, data sources work via the graph while functions do not.